Feature/new test script

dlsk0303 opened the pull request:
TestShell 의 run() 에서 찾아지지 않는 command 일 때, DLL 로드 후 execute() 호출하도록 구현했습니다.

남은 것 :
1. TestScript_01.dll, TestScript_02.dll 식으로 여러 DLL 들을 loop 에서 찾는 식으로 변경.
2. 현재 콜백을  main 에서 하는데 TestShell 생성자로 옮기기.

리뷰 부탁드립니다.

yn0303 committed:
first commit

yn0303 committed:
addShell() calling success

yn0303 committed:
When the 'write' command is entered, the command buffer log output is confirmed

yn0303 committed:
Green) The TestShell class has been modified to include a TestScript object as a member.

dlsk0303 committed:
Merge branch 'master' into feature/newTestScript
Test Script modified.
객체 전달	TestScriptCallback* 구조체 포인터 사용
객체 생명주기	EXE에서 생성 → DLL에서 참조만 함
DLL 함수 호출	EXE에서 execute() 호출
객체 간 의존 제거	DLL은 EXE 내부 객체를 전혀 모르도록 구성

dlsk0303 committed:
Reverted the main function to its original form.

dlsk0303 committed:
Refactoring) Cleaned up log output

1. Removed unnecessary code
2. Added usage guide for callback functions
3. Replaced cout with LOG_PRINT

dlsk0303 committed:
Refactoring) execute function does nothing for now

dlsk0303 committed:
Refactoring) Performed additional code cleanup

yn0303 committed:
Green) New TestScript created

yongjae410 commented on TestScript/TestScript.cpp:
```diff
@@ -0,0 +1,194 @@
+            result = fullWriteAndReadCompare();
+            printTestScriptResult(result);
+    }
+#endif
```
사용하지 않을 코드라면 #if 0 로 감싼 코드는 삭제하는게 좋을 것 같습니다

dlsk0303 replied:
네, 아래 함수 호출 시 아무것도 안하고  -1 리턴하는 것으로 해두겠습니다.
 int TestScript::execute(string command)

yongjae410 commented on TestScript/TestScript.cpp:
```diff
@@ -0,0 +1,194 @@
+    return result;
+}
+
+int TestScript::fullWriteAndReadCompare() {
```
test script #1 테스트를 위한 함수인 걸로 보여서 사용하지 않는 함수일 것 같은데 맞을까요?
맞다면 삭제 부탁드립니다

yongjae410 commented on TestScript/TestScript.cpp:
```diff
@@ -0,0 +1,194 @@
+    return result;
+}
+
+int TestScript::eraseAndWrite() {
```
이 함수부터 아래로 있는 함수들 모두 다 테스트 용도인 것 같은데 맞나요?
TestScript 5번을 fullRead 라고 가정하고 구현한 것으로 하고, 모두 삭제해도 괜찮을 것 같아요

dlsk0303 replied:
네~ 모두 삭제하겠습니다.

yongjae410 commented on TestScript/framework.h:
```diff
@@ -0,0 +1,5 @@
+
+#define WIN32_LEAN_AND_MEAN             // 거의 사용되지 않는 내용을 Windows 헤더에서 제외합니다.
+// Windows 헤더 파일
+#include <windows.h>
```
이 헤더는 왜 추가되는건지 알 수 있을까요?

dlsk0303 replied:
dll main 만들어지면서 자동으로 생성되었는데 빼도 빌드 되는지 체크해볼께요.

dlsk0303 replied:
말씀하신 부분 반영하여 push 다시 했습니다.

yongjae410 commented on TestShell/TestShell.h:
```diff
@@ -8,16 +8,21 @@
-        : reader(reader), writer(writer), eraser(eraser), flusher(flusher) {}
+    TestShell(SSDExecutor* reader, SSDExecutor* writer, SSDExecutor* eraser, SSDExecutor* flusher, TestScriptCallback* cb)
+        : reader(reader), writer(writer), eraser(eraser), flusher(flusher) {
+        cb_ = cb;
```
코딩 스타일을 통일하는 관점에서
`: reader(reader), writer(writer), eraser(eraser), flusher(flusher), cb_(cb) { }` 로 구현하는건 어떨까요?

yn0303 committed:
Remove Test Code in TestScript and Modify to register the callback in the TestShell constructor.

yn0303 committed:
Refactoring) First, check for TestScript.dll, and if it's not found, then try loading TestScript_0.dll, TestScript_1.dll, ..., TestScript_9.dll.

yongjae410 commented on TestScript/TestScript.h:
```diff
@@ -0,0 +1,36 @@
+    int runSSDWriter(int lba, std::string& data, const int& numOfTimes, bool print2Console);
+    void printTestScriptResult(int result);
+    int fullWriteAndReadCompare();
+    int readCompare(string& expected);
```
cpp 에 맞춰서 헤더파일도 정리가 필요해 보여요

dlsk0303 replied:
수정해서 다시 올렸습니다.

yongjae410 commented on TestShell/TestShell.h:
```diff
@@ -1,4 +1,4 @@
-#pragma once
+﻿#pragma once
```
pragma once 가 두 줄 생기는 것 같은데 확인 부탁드려요

dlsk0303 replied:
수정해서 다시 올렸습니다.

yongjae410 commented on TestShell/TestShell.cpp:
```diff
@@ -1,9 +1,21 @@
+TestShell::TestShell(SSDExecutor* reader, SSDExecutor* writer, SSDExecutor* eraser, SSDExecutor* flusher)
+    : reader(reader), writer(writer), eraser(eraser), flusher(flusher) {
+    // 2. TestScript에 등록할 콜백 구조체 세팅
+    static TestScriptCallback cb;
```
이건 멤버변수 cb_ 에 넣는 객체인데, static 으로 정의하신 이유가 있을까요?

dlsk0303 replied:
TestScriptCallback cb; 를 private 멤버로 갖도록 수정했습니다.

yn0303 committed:
Refactoring) To fix the DLL loading bug

Try to find TestScript.dll first, and if it is not found,
then search for TestScript_0.dll, TestScript_1.dll, ...,
up to TestScript_9.dll and load the DLL.

yn0303 committed:
Modified to return since it was successful.

yongjae410 reviewed (approved):
확인했습니다~

issuxark11 reviewed (approved):
확인했습니다
